src/codec/encoder/PyEncoder.cpp

```diff
@@ -66,6 +67,89 @@ const std::wstring PyEncoder::ID = L"com.esri.pyprt.PyEncoder";
 const std::wstring PyEncoder::NAME = L"Python Geometry and Report Encoder";
 const std::wstring PyEncoder::DESCRIPTION = L"Encodes geometry and CGA report for Python.";
 
+/**
+ * Manage the reports collection.
+ */
+void PyEncoder::processReports(prtx::GenerateContext& context, size_t initialShapeIndex, IPyCallbacks* cb) {

```

Comment:
Nice cleanup of the ::encode function.

As this function does not modify the PyEncoder class state, it should be private const-qualified, or better private static, or even better moved out of the class ant put as a free function into the anonymous namespace in the cpp.